integration/spark/spark3/src/main/java/io/openlineage/spark3/agent/utils/PlanUtils3.java

```diff
@@ -101,10 +101,9 @@ public static <D extends OpenLineage.Dataset> List<D> fromDataSourceV2Relation(
       OpenLineageContext context,
       DataSourceV2Relation relation,
       OpenLineage.DatasetFacetsBuilder datasetFacetsBuilder) {
-
     if (relation.identifier().isEmpty()) {
-      throw new IllegalArgumentException(
-          "Couldn't find identifier for dataset in plan " + relation);
+      log.warn("Couldn't find identifier for dataset in plan {}", relation);
```

Comment:
@collado-mike, @pawel-big-lebowski: Do we know why we throw an exception when invoking [`PlanUtils3. fromDataSourceV2Relation()`](https://github.com/OpenLineage/OpenLineage/blob/main/integration/spark/spark3/src/main/java/io/openlineage/spark3/agent/utils/PlanUtils3.java#L106), but in [`DatasetVersionDatasetFacetUtils`](https://github.com/OpenLineage/OpenLineage/blob/main/integration/spark/spark3/src/main/java/io/openlineage/spark3/agent/utils/DatasetVersionDatasetFacetUtils.java#L35) we log a warning? Throwing an exception means we stop traversing the plan and "dropping" the event. By logging a warning, then return an empty list we can continue the spark plan traversal.

Reply:
Seems we should log a warning in both cases - I feel that's more in line with how we treat nodes we simply don't know how to handle.

Reply:
Ah nice, might have just been an oversight originally 👍